lms/djangoapps/teams/views.py

```diff
@@ -541,6 +561,15 @@ def get_queryset(self):
         """Returns the queryset used to access the given team."""
         return CourseTeam.objects.all()
 
+    def delete(self, request, team_id):
+        """DELETE /api/team/v0/teams/{team_id}"""
+        team = get_object_or_404(CourseTeam, team_id=team_id)
+        self.check_object_permissions(request, team)
+        # Note: also deletes all team memberships associated with this team
```

Comment:
Yes, Django automatically propagates deletes to anything with a foreign key to the deleted model. https://docs.djangoproject.com/en/1.4/ref/models/querysets/#delete